pkg/unpackerr/cnfgfile.go

```diff
@@ -311,12 +312,60 @@ func (u *Unpackerr) validateApp(conf *StarrConfig, app starr.App) error {
 		conf.Protocols = defaultProtocol
 	}
 
+	// Configure TLS and HTTP client
+	tlsConfig, err := u.configureTLS(conf, app)
+	if err != nil {
+		return err
+	}
+
 	conf.Config.Client = &http.Client{
 		Timeout: conf.Timeout.Duration,
 		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: !conf.ValidSSL}, //nolint:gosec
+			TLSClientConfig: tlsConfig,
 		},
 	}
 
 	return nil
 }
+
+// configureTLS creates and configures the TLS config for mTLS support.
+func (u *Unpackerr) configureTLS(conf *StarrConfig, app starr.App) (*tls.Config, error) {
+	// Create TLS config - default behavior unchanged
+	tlsConfig := &tls.Config{InsecureSkipVerify: !conf.ValidSSL} //nolint:gosec
+
+	// Add mTLS if certificates are configured
+	if conf.TLSClientCert != "" && conf.TLSClientKey != "" {
```

Comment:
```suggestion
	if conf.TLSClientCert != "" || conf.TLSClientKey != "" {
```

I usually do an "or" on these two variables when dealing with SSL. In case a user sets one, they'll get an error. Instead of nothing.